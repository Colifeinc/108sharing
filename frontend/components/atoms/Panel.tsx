import * as React from 'react'
import classNames from 'classnames'

interface IPanelProps {
  title?: string | React.ReactElement
  className?: string
  wrapperFull?: boolean
  contentFull?: boolean
}

const Panel: React.FC<IPanelProps> = ({
  title,
  children,
  className = '',
  contentFull,
  wrapperFull,
}) => {
  return (
    <div
      className={classNames([
        'm-auto px-6 py-10 rounded-xlg shadow-colored-lg bg-white',
        wrapperFull ? '' : 'max-w-4xl',
        className,
      ])}
    >
      <div className={classNames(['m-auto', contentFull ? '' : 'max-w-2xl'])}>
        {title && <h1 className="mb-8 text-xl font-bold">{title}</h1>}
        <div>{children}</div>
      </div>
    </div>
  )
}

export default Panel
