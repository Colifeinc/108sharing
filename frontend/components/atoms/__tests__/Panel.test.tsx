import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import * as React from 'react'
import Panel from '../Panel'

describe('atoms/Panel.tsx', () => {
  const props = {
    title: 'Title',
  }

  it('should matches to the snapshot', () => {
    const wrapper = shallow(<Panel {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('should renders children when passed in', () => {
    const wrapper = shallow(
      <Panel>
        <div className="unique" />
      </Panel>
    )
    expect(wrapper.contains(<div className="unique" />)).toBe(true)
  })
})
