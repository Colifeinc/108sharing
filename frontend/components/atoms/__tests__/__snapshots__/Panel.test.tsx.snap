// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms/Panel.tsx should matches to the snapshot 1`] = `
<div
  className="m-auto px-6 py-10 rounded-xlg shadow-colored-lg bg-white max-w-4xl"
>
  <div
    className="m-auto max-w-2xl"
  >
    <h1
      className="mb-8 text-xl font-bold"
    >
      Title
    </h1>
    <div />
  </div>
</div>
`;
